token/services/db/sql/common/query.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package common
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	SelectStatement         = `SELECT`
+	SelectDistinctStatement = `SELECT DISTINCT`
+	InsertStatement         = `INSERT INTO`
+	UpdateStatement         = `UPDATE`
+)
+
+type Select struct {
+	stmt    string
+	columns []string
+	from    []string
+	where   string
+	orderBy string
+}
+
+func NewSelect(columns ...string) *Select {
+	return &Select{
+		stmt:    SelectStatement,
+		columns: columns,
+	}
+}
+
+func NewSelectDistinct(columns ...string) *Select {
+	return &Select{
+		stmt:    SelectDistinctStatement,
+		columns: columns,
+	}
+}
+
+func (s *Select) From(tables ...string) *Select {
+	s.from = tables
+	return s
+}
+
+func (s *Select) Where(where string) *Select {
+	s.where = where
+	return s
+}
+
+func (s *Select) Compile() (string, error) {
+	sb := new(strings.Builder)
+	sb.WriteString(s.stmt)
+	sb.WriteString(" ")
+	if len(s.columns) > 0 {
+		sb.WriteString(strings.Join(s.columns, ","))
+		sb.WriteString(" ")
+	}
+	if len(s.from) > 0 {
+		sb.WriteString("FROM ")
+		sb.WriteString(strings.Join(s.from, " "))
+		sb.WriteString(" ")
+	}
+	if len(s.where) > 0 {
+		if !strings.HasPrefix(s.where, "WHERE") {
+			sb.WriteString(" WHERE ")
+		}
+		sb.WriteString(s.where)
+	}
+	if len(s.orderBy) > 0 {
+		if !strings.HasPrefix(s.orderBy, "ORDER BY") {
+			sb.WriteString(" ORDER BY ")
+		}
+		sb.WriteString(s.orderBy)
+	}
+	return sb.String(), nil
+}
+
+func (s *Select) OrderBy(orderBy string) *Select {
+	s.orderBy = orderBy
+	return s
+}
+
+type Insert struct {
+	stmt  string
+	rows  string
+	table string
+}
+
+func NewInsertInto(table string) *Insert {
+	return &Insert{
+		stmt:  InsertStatement,
+		table: table,
+	}
+}
+
+func (i *Insert) Rows(rows string) *Insert {
```

Comment:
This refers to fields/columns I think. Also we could pass a slice to make it clear.